sdk/python/lib/pulumi/x/automation/stack.py

```diff
@@ -0,0 +1,532 @@
+import grpc
+from concurrent import futures
+from enum import Enum
+from datetime import datetime
+from typing import List, Any, Mapping, MutableMapping, Optional
+
+from .cmd import CommandResult, _run_pulumi_cmd, OnOutput
+from .config import ConfigValue, ConfigMap, _SECRET_SENTINEL
+from .errors import StackAlreadyExistsError, CommandError
+from .server import LanguageServer
+from .workspace import Workspace, PulumiFn
+from ...runtime.settings import _GRPC_CHANNEL_OPTIONS
+from ...runtime.proto import language_pb2_grpc
+
+_DATETIME_FORMAT = '%Y-%m-%dT%H:%M:%S.%fZ'
+
+
+class ExecKind(str, Enum):
+    LOCAL = "auto.local"
+    INLINE = "auto.inline"
+
+
+class StackInitMode(Enum):
+    CREATE = "create"
+    SELECT = "select"
+    CREATE_OR_SELECT = "create_or_select"
+
+
+class OutputValue:
+    value: Any
+    secret: bool
+
+    def __init__(self, value: Any, secret: bool):
+        self.value = value
+        self.secret = secret
+
+    def __repr__(self):
+        return _SECRET_SENTINEL if self.secret else repr(self.value)
+
+
+OutputMap = MutableMapping[str, OutputValue]
+
+OpMap = MutableMapping[str, int]
+
+
+class UpdateSummary:
+    # pre-update info
+    kind: str
+    start_time: datetime
+    message: str
+    environment: Mapping[str, str]
+    config: ConfigMap
+
+    # post-update info
+    result: str
+    end_time: datetime
+    version: Optional[int]
+    deployment: Optional[str]
+    resource_changes: Optional[OpMap]
+
+    def __init__(self,
+                 kind: str,
+                 start_time: datetime,
+                 message: str,
+                 environment: Mapping[str, str],
+                 config: Mapping[str, dict],
+                 result: str,
+                 end_time: datetime,
+                 version: Optional[int] = None,
+                 deployment: Optional[str] = None,
+                 resource_changes: Optional[OpMap] = None):
+        self.kind = kind
+        self.start_time = start_time
+        self.end_time = end_time
+        self.message = message
+        self.environment = environment
+        self.result = result
+        self.Deployment = deployment
+        self.resource_changes = resource_changes
+        self.version = version
+        self.config: ConfigMap = {}
+        for key in config:
+            self.config[key] = ConfigValue(**config[key])
+
+    def __repr__(self):
+        return f"UpdateSummary(result={self.result!r}, version={self.version!r}, start_time='{self.start_time!s} UTC', end_time='{self.end_time!s} UTC')"
+
+
+class BaseResult:
+    stdout: str
+    stderr: str
+    summary: UpdateSummary
+
+    def __init__(self, stdout: str, stderr: str, summary: UpdateSummary):
+        self.stdout = stdout
+        self.stderr = stderr
+        self.summary = summary
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}(result={self.summary.result!r}, start_time='{self.summary.start_time!s} UTC', end_time='{self.summary.end_time!s} UTC')"
```

Comment:
This doesn't look right (this class doesn't have `result`, `start_time`, etc. args).